src/server/common/model/answer/autocomplete/autocomplete.js

```diff
@@ -0,0 +1,148 @@
+import Joi from 'joi'
+import { AnswerModel } from '../answer-model.js'
+import { validateAnswerAgainstSchema } from '../validation.js'
+import { NotImplementedError } from '../../../helpers/not-implemented-error.js'
+import { escapeHtml } from '../../../helpers/escape-text.js'
+
+/** @import {AnswerViewModelOptions} from '../answer-model.js' */
+
+/**
+ * @param {AutocompleteConfig} config
+ * @returns {Joi.Schema}
+ */
+const textSchema = ({ payloadKey, validation }) => {
+  const stringValidation = Joi.string().trim().required()
+
+  const messages = {
+    'any.required': validation.empty?.message ?? '',
+    'string.empty': validation.empty?.message ?? ''
+  }
+
+  return Joi.object({
+    [payloadKey]: stringValidation.messages(messages)
+  })
+}
+
+/**
+ * export @typedef {{
+ *  payloadKey: string,
+ *  characterWidth?: 2 | 4 | 10 | 20,
+ *  isPageHeading? : boolean,
+ *  hint?: string,
+ *  validation: {
+ *    empty?: { message: string },
+ *  }
+ *  items: {
```

Comment:
In the other model classes, we use an object whose keys are the values.  Would be good to follow suit: 

```javascript
 /**
 * @typedef {{ label: string, hint?: string }} RadioOption
 * @typedef {'inline' | 'stacked'} RadioButtonLayout
 * export @typedef {{
 *  payloadKey: string,
 *  hint?: string,
 *  options: Record<string, RadioOption>,
 *  validation: {
 *    empty: string
 *  }
 * layout?: RadioButtonLayout
 * }} RadioButtonConfig
 */
```